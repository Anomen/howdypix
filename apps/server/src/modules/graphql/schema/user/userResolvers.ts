import { NexusGenFieldTypes } from "@howdypix/graphql-schema/schema";
import { ApolloContext } from "../../../../types";

export const getCurrentUserResolver = () => (
  root: {},
  args: {},
  ctx: ApolloContext
): NexusGenFieldTypes["Query"]["getCurrentUser"] => ctx.user;
