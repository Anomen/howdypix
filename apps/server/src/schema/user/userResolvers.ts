import { NexusGenFieldTypes } from "@howdypix/graphql-schema/schema";
import { UserConfigState } from "../../state";
import { createTransport } from "nodemailer";
import smtpTransport from "nodemailer-smtp-transport";
import { magicLink } from "../../email";
import { appDebug } from "@howdypix/utils";
import { NexusGenArgTypes } from "@howdypix/graphql-schema/schema";
import { generateCode, isEmailValid, storeCode } from "../../lib/auth";
import { ApolloContext } from "../../types";

const debug = appDebug("gql:user");

export const currentUserResolver = () => async (
  root: {},
  args: {},
  ctx: ApolloContext
) =>
  new Promise<NexusGenFieldTypes["Query"]["currentUser"]>(async resolve => {
    resolve(ctx.user || undefined);
  });
