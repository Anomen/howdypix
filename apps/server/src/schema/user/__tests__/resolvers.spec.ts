import { getCurrentUserResolver } from "../userResolvers";

describe("currentUserResolver", () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  const resolver = getCurrentUserResolver();

  test("should return the current user in the context", async () => {
    expect(
      resolver({}, {}, { user: { name: "name", email: "email" } })
    ).toMatchSnapshot();
  });

  test("should return null if no user in the context", async () => {
    expect(resolver({}, {}, { user: null })).toMatchSnapshot();
  });
});
