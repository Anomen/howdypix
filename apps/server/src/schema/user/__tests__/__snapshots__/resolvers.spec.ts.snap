// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`currentUserResolver should return null if no user in the context 1`] = `null`;

exports[`currentUserResolver should return the current user in the context 1`] = `
Object {
  "email": "email",
  "name": "name",
}
`;
